explorer/explorerroutes.go

```diff
@@ -224,7 +226,114 @@ func (exp *explorerUI) StakeDiffWindows(w http.ResponseWriter, r *http.Request)
 		int64(rows),
 		exp.Version,
 		exp.NetName,
+		exp.MempoolData,
 	})
+	exp.MempoolData.RUnlock()
+
+	if err != nil {
+		log.Errorf("Template execute failure: %v", err)
+		exp.StatusPage(w, defaultErrorCode, defaultErrorMessage, ErrorStatusType)
+		return
+	}
+
+	w.Header().Set("Content-Type", "text/html")
+	w.WriteHeader(http.StatusOK)
+	io.WriteString(w, str)
+}
+
+// DayBlocksListing handles "/day" page.
+func (exp *explorerUI) DayBlocksListing(w http.ResponseWriter, r *http.Request) {
+	exp.timeBasedBlocksListing("Days", w, r)
+}
+
+// WeekBlocksListing handles "/week" page.
+func (exp *explorerUI) WeekBlocksListing(w http.ResponseWriter, r *http.Request) {
+	exp.timeBasedBlocksListing("Weeks", w, r)
+}
+
+// MonthBlocksListing handles "/month" page.
+func (exp *explorerUI) MonthBlocksListing(w http.ResponseWriter, r *http.Request) {
+	exp.timeBasedBlocksListing("Months", w, r)
+}
+
+// YearBlocksListing handles "/year" page.
+func (exp *explorerUI) YearBlocksListing(w http.ResponseWriter, r *http.Request) {
+	exp.timeBasedBlocksListing("Years", w, r)
+}
+
+// TimeBasedBlocksListing is the main handler for "/day", "/week", "/month" and "/year".
+func (exp *explorerUI) timeBasedBlocksListing(val string, w http.ResponseWriter, r *http.Request) {
+	if exp.liteMode {
+		exp.StatusPage(w, fullModeRequired,
+			"Time based blocks listing page cannot run in lite mode.", NotSupportedStatusType)
+	}
+
+	grouping := dbtypes.TimeGroupingFromStr(val)
```

Comment:
Please pass a `TimeBasedGrouping` instead of a string to `timeBasedBlocksListing`.  This should be simpler, and will eliminate the potential future bug where TimeGroupingFromStr and the strings in the functions like `YearBlocksListing` may no longer agree.

Reply:
The string is needed on the front end since the four routes share a template. I could pass the string and the `TimeBasedGrouping` object but that might result to duplication and also create bugs when there are inconsistencies between the two values set.

Reply:
The nice thing about objects with methods is that you can call the method from the template.  :D  I'm referring to `func (g TimeBasedGrouping) String() string`.  In fact, since this type implements the `Stringer` interface, I'd be surprised if you even need to call String() manually, but you can.

Reply:
This can be revisited, but please note by above comment.

Reply:
This function `func (g TimeBasedGrouping) String() string` is going to be very important in relation to the next PR that I am currently testing. 
Comment noted though.